### PR TITLE
Add classes: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by eric2umeh**
On this PR,

-[x] Create a `person.rb`, `student.rb` & `teacher.rb` files following this [instructions](http://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md)
-[x] Add Linters
-[x] Follow correct Gitflow and best practice for Ruby

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Person
+  def initialize(age, name = 'Unknown', _parent_permision: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+  end
+
```
Comment (julie-ify) [inline person.rb]: - [ ] Kindly add the `@parent_permission` instance variable here as well.

  ```
  ...
  @parent_permission = parent_permission
  ```

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || parent_permission?
```
Comment (julie-ify) [inline person.rb]: - [ ] Kindly use the `@parent_permission` variable here instead of just `parent_permission`.

File: student.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Student < Person
+  def initialize(age, classroom, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
Comment (julie-ify) [inline student.rb:5]: - [ ] You are using the colon notation for `parent_permission`. This means the way of assigning it in the `super()` method would be different.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
```
Comment (julie-ify) [inline teacher.rb:5]: - [ ] You are using the colon notation for `parent_permission`. This means the way of assigning it in the `super()` method would be different.

  ```
  super(age, name, parent_permission: parent_permission)
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @eric2umeh,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 4 above)
Comment (eric2umeh) [reply teacher.rb:5]: Hey @julie-ify 
Thank you so much for the corrections but more importantly how you gave solutions. I am so grateful and Happpy, ha

File: student.rb (hunk 3 above)
Comment (ShadyShawkat) [inline student.rb]: - [ ] As remarked by the previous reviewer, kindly change the `super` method call to the following, as you're using the colon notation for `parent_permission`.

```ruby
super(age, name, parent_permission: parent_permission)
```

File: teacher.rb (hunk 4 above)
Comment (ShadyShawkat) [inline teacher.rb]: - [ ] As remarked by the previous reviewer, kindly change the `super` method call to the following, as you're using the colon notation for `parent_permission`.

```ruby
super(age, name, parent_permission: parent_permission)
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @eric2umeh, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Image

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — APPROVED** [review]
Hi @eric2umeh 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Proper use of Gitflow 👍 
* Classes and Objects implemented 👍 
* Good looking readme 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**